build-tools/automation/azure-pipelines.yml

```diff
@@ -0,0 +1,57 @@
+# Xamarin.Android Pipeline
+
+trigger:
+  - master
+  - d16*
+
+stages:
+- stage: Build
+  jobs:
+  - job: Mac
+    pool: Hosted Mac Internal
+    timeoutInMinutes: 240
+    cancelTimeoutInMinutes: 5
+    variables:
+      configuration: Release
+      msbuildArgs: /p:AutoProvision=True /p:AutoProvisionUsesSudo=True /p:IgnoreMaxMonoVersion=False
+      JavaSdkDirectory: /Library/Java/JavaVirtualMachines/jdk1.8.0_144.jdk/Contents/Home/
+      JAVA_HOME: $(JavaSdkDirectory)
+    steps:
+    - script: make prepare-external-git-dependencies CONFIGURATION=$(configuration)
+      displayName: make prepare-commercial
+      condition: eq(variables['XA.Commercial.Build'], 'true')
+      env:
+        GH_AUTH_SECRET: $(Github.Token)
+    - task: provisionator@2
+      condition: and(succeeded(), eq(variables['XA.Commercial.Build'], 'true'))
+      inputs:
+        provisionator_uri: $(provisionator-uri)
+        github_token: $(GitHub.Token)
+        provisioning_script: $(System.DefaultWorkingDirectory)/external/monodroid/build-tools/provisionator/profile.csx
+        provisioning_extra_args: -vv
+    - script: make prepare-deps V=1 CONFIGURATION=$(configuration) MSBUILD_ARGS="$(msbuildArgs)"
+      displayName: make prepare-deps
+    - script: make prepare jenkins V=1 CONFIGURATION=$(configuration) MSBUILD_ARGS="$(msbuildArgs)"
+      displayName: make prepare jenkins
+    - script: make create-installers V=1 CONFIGURATION=$(configuration)
+      displayName: make create-installers
+    - script: |
+       mkdir -p $(System.DefaultWorkingDirectory)/unsigned-installers
+       cp bin/Build$(configuration)/*.vsix $(System.DefaultWorkingDirectory)/unsigned-installers
+       cp bin/Build$(configuration)/*.pkg $(System.DefaultWorkingDirectory)/unsigned-installers
+       cp *.zip $(Build.ArtifactStagingDirectory)
+      displayName: copy build artifacts
+    - task: PublishPipelineArtifact@0
+      displayName: upload unsigned installers
+      inputs:
+        artifactName: unsigned
+        targetPath: $(System.DefaultWorkingDirectory)/unsigned-installers
+    - script: cp bin/*/*.binlog $(Build.ArtifactStagingDirectory)
```

Comment:
Historically speaking, we frequently need *far* more than just the `.binlog` files.  This would be a case for "migrating" `build-tools/scripts/Packaging.mk`/`make package-build-status` into MSBuild-land.

Reply:
I've moved these make rules into a new msbuild targets file which is imported by Xamarin.Android.Tools.BootstrapTasks.csproj here - https://github.com/xamarin/xamarin-android/pull/3085/commits/e839224ef039688a98ea297beac83f8f2df36436. I was questioning the best place to put these targets, but I figured this location was somewhat sensible as the new targets use the `Zip` task which belongs to this project.